### PR TITLE
feat(appsync): add authorization config to the HttpDataSource

### PR DESCRIPTION
**Pull request by haruharuharuby**
This PR supersedes #9971 

### Description
Adding authorization config to HttpDatasource in aws-appsync module.
Users will be able to specify the awsIamConfig in httpConfig as parameter of HttpDataSource.

Fixes: #9934 

----

### CODE REVIEW

File: packages/@aws-cdk/aws-appsync/README.md [hunk 1]
```diff
@@ -82,6 +84,83 @@ demoDS.createResolver({
...
+    },
+    "body": {
+      "stateMachineArn": "<your step functions arn>",
+      "input": "{ \"id\": \"$context.arguments.id\" }"
+    }
+  }
+}
+```
+
+GraphQL request mapping template `response.vtl`:
```
Comment (BryanPan342) [inline packages/@aws-cdk/aws-appsync/README.md:125]:
```suggestion
GraphQL response mapping template `response.vtl`:
```

File: packages/@aws-cdk/aws-appsync/README.md [hunk 2]
```diff
@@ -82,6 +84,83 @@ demoDS.createResolver({
...
+```
+
+CDK stack file `app-stack.ts`:
+
+```ts
+import * as appsync from '@aws-cdk/aws-appsync';
+
+const api = new appsync.GraphQLApi(scope, 'id', {
+  name: 'api',
+  schema: appsync.Schema.fromFile(join(__dirname, 'schema.graphql))
```
Comment (BryanPan342) [inline packages/@aws-cdk/aws-appsync/README.md:140]:
```suggestion
  schema: appsync.Schema.fromFile(join(__dirname, 'schema.graphql')),
```

File: packages/@aws-cdk/aws-appsync/README.md [hunk 3]
```diff
@@ -82,6 +84,83 @@ demoDS.createResolver({
...
+  "id": "${context.result.id}"
+}
+```
+
+CDK stack file `app-stack.ts`:
+
+```ts
+import * as appsync from '@aws-cdk/aws-appsync';
+
+const api = new appsync.GraphQLApi(scope, 'id', {
```
Comment (BryanPan342) [inline packages/@aws-cdk/aws-appsync/README.md:138]:
```suggestion
const api = new appsync.GraphqlApi(scope, 'api', {
```

File: packages/@aws-cdk/aws-appsync/README.md [hunk 4]
```diff
@@ -18,8 +18,10 @@ APIs that use GraphQL.
 
 ### Example
 
+#### DynamoDB
```
Comment (BryanPan342) [inline packages/@aws-cdk/aws-appsync/README.md:21]:
```suggestion
### DynamoDB
```

File: packages/@aws-cdk/aws-appsync/README.md [hunk 5]
```diff
@@ -128,8 +207,7 @@ const api = appsync.GraphqlApi(stack, 'api', {
 });
 ```
 
-### Imports
-
+## Imports
```
Comment (BryanPan342) [inline packages/@aws-cdk/aws-appsync/README.md:210]:
```suggestion
### Imports
```

**Review by BryanPan342 — COMMENTED** [review]
Final touches!

**Comment by haruharuharuby** [comment]
@BryanPan342 thank you final check!

**Review by BryanPan342 — DISMISSED** [review]
LGTM 🎉

Thanks for the contribution @haruharuharuby 😊

**Review by BryanPan342 — APPROVED** [review]
There was a weird bug in s3-deployment but this should work :)

**Comment by haruharuharuby** [comment]
Thanks again many supports. :) @BryanPan342

**Comment by andrestone** [comment]
Does anyone know what are the values expected by `signingServiceName`? It's completely obscure if this refers to the ARN standard or the endpoint subdomain. No clue at the documentation either.

**Pull request merged**